lib/Dialect/mhlo/transforms/mhlo_fusion.cc

```diff
@@ -545,6 +545,38 @@ struct MhloFusionPass : public MhloFusionPassBase<MhloFusionPass> {
       for (Value v : outputs) {
         output_types.push_back(v.getType());
       }
+      //      /-----\
+      //     /       V
+      // A(fused) -- B -- C(fused) -- D(fused)
+      // mlir ops Adjacency List likely above
+      // the B is consumer of fused A, so B need move behind D
+      // because fusion op create at D's location
+      DenseSet<Operation*> fused_set(pattern.begin(), pattern.end());
+      DenseSet<Operation*> consumers_set;
+      SmallVector<Operation*, 4> consumers_vec;
+      auto first_iter = pattern.front()->getIterator();
+      auto last_iter = pattern.back()->getIterator();
+      while (first_iter != last_iter) {
+        // isn't fused op && consumer's op
+        // move this after fusion op
+        Operation* cur_op = &*first_iter;
+        if (fused_set.end() == fused_set.find(cur_op)) {
```

Comment:
I think there is a `contain` method? 
```suggestion
        if (!fused_set.contains(cur_op)) {
```
(same below)

Reply:
thanks update later